single_include/kompute/Kompute.hpp

```diff
@@ -154,13 +156,125 @@ class Shader {
      * @param source An individual raw glsl shader in string format
      * @param entryPoint The function name to use as entry point
      * @param definitions List of pairs containing key value definitions
+     * @param resourcesLimit A list that contains the resource limits for the GLSL compiler
      * @return The compiled SPIR-V binary in unsigned int32 format
      */
     static std::vector<uint32_t> compile_source(
             const std::string& source,
             const std::string& entryPoint = "main",
-            std::vector<std::pair<std::string,std::string>> definitions = {});
+            std::vector<std::pair<std::string,std::string>> definitions = {},
+            const TBuiltInResource& resources = defaultResource);
 
+private:
+    // The default resource limit for the GLSL compiler, can be overwritten
+    // Has been adobted by:
+    // https://github.com/KhronosGroup/glslang/blob/master/StandAlone/ResourceLimits.cpp
+    static constexpr TBuiltInResource defaultResource = {
```

Comment:
here you can just call it `const` as that is the same as `constexpr`

Reply:
It yielded me an error, this:
`constexpr' needed for in-class initialization of static data member`
more on this here:
https://stackoverflow.com/questions/9141950/initializing-const-member-within-class-declaration-in-c

Reply:
Ok i see, although not clear why it would error being static inside a class, may be worth having a deeper look, as the rest of the functions are within the Shader scope so would be best to have this variable also within

Reply:
Yes, I'm trying to do so for hours. The closet thing I have come into is `static constexpr TBuiltInResource const& defaultResource`(got it from [here](https://en.cppreference.com/w/cpp/language/constexpr)), but instead of the undefined reference I got is `multiple definition of `_ZGRN2kp6Shader15defaultResourceE_'; CMakeFiles/test_kompute.dir/TestAsyncOperations.cpp.o:(.rodata+0x0): first defined her`. Any ideas?

Reply:
Hmm ok I see, I think given this is taking too long let's just get it as global and we can move it (we an open an issue separate)

Reply:
Yeah sounds good